doc/user_guide/user_guide.md

```diff
@@ -0,0 +1,151 @@
+        else
+            return loader
+        end
+    end
+)
+```
+
+## Using `remotelog`
+
+### Loading the Module
+
+The first thing you need is obviously a `require` instruction. In the simplest possible case, that is:
+
+```lua
+local log = require("remotelog")
+```
+
+### Configuring `remotelog`
+
+There are two ways to configure the module. One is regular configuration through setters.
+
+The other one is through an `init()` function. You will rarely ever need the second one. It is intended for one-time settings and mainly exists to make unit testing easier. You can control the timestamp pattern and use of the high-resolution timer through `init()`.
+
+While this is optional, we recommend using `set_client_name()` before your first log message, so that you get nice environment information at the beginning of each log. The client name should identify the application that uses the log and it's version number. But you could also write any other environment information into it that you want to appear right in front of the log.
+
+```lua
+log.set_client_name("my-script 1.3.0")
+```
+
+`remotelog` supports the following levels listed below. The default log level is `INFO`.
+
+<dl>
+<dt>NONE</dt><dd>no log entries are generated on this level.</dd>
+<dt>FATAL</dt><dd>should be used for in cases of errors that require immediate program termination.</dd>
+<dt>ERROR</dt><dd>is best used in cases where the calling code should be given a chance to deal with the error or at least clean up before termination.</dd>
+<dt>WARN</dt><dd>stands for situations that don't have an immediate adverse effect, but might cause an error later.</dd>
+<dt>INFO</dt><dd>is for sparse log information that you want to appear in a standard log.</dd>
+<dt>CONFIG</dt><dd>should be used to log parameters and environment of the client.</dd>
+<dt>DEBUG</dt><dd>is targeted at developers when trying to analyze program flow or problems.</dd>
+<dt>TRACE</dt><dd>is showing details of internal state.</dd>
+</dl>
+
+You can change the log level with the function `set_level()`:
+
+```lua
+log.set_level("DEBUG")
+```
+
+### Writing log Messages
+
+Writing log messages is as easy as it gets:
+
+```lua
+log.info("Initialization complete.")
+```
+
+If your provide more than one parameter, `exsollog` will interpret the first one as a format string (see [`string.format()`](https://www.lua.org/manual/5.1/manual.html#pdf-string.format) from the Lua standard library). All other parameters are then taken as values for the placeholders inside that string.
+
+```lua
+log.debug("Finished task '%s' after %07.3dms", task_name, elapsed_time)
+```
+
+### Connecting to a Remote log Receiver
+
+The most important part of `remotelog` is that it supports remote logging. Using that feature is trivial.
+
+To try it out, first start a socket listener on a remote machine, for example using `netcat` on port 3000:
+
+```bash
+nc -lkp 3000
```

Comment:
Could we maybe suggest in addition a more convenient log viewer. Unfortunately, I don't know any. On android I used logcat. Filtering the logs using a gui makes it so much easier...